### ISSUE TITLE
How to combine the monolingual side of all the benchmarks?

### ISSUE THREAD

**Issue by look4pritam** [AI4Bharat/IndicTrans2#18]
How to combine the monolingual side of all the benchmarks?
Is it similar to combining different sources for training?

How much time is required to train the eng to Indic model using four A100 graphic cards?

**Comment by jaygala24** [comment]
Hey

In order to combine the monolingual side of all benchmarks, simply iterate through the language-specific test set files one by one for each language across all benchmarks, read the lines from each file, merge them together, and then save the merged content into a designated output file. I think this should be straightforward to implement in python3.

Based on our estimates, training the English-Indic model using 4 A100 40GB GPUs will likely take approximately 45 days with a training dataset of 640 million words (BPCC + Backtranslated data). However, we would recommend you use the [model released](https://github.com/AI4Bharat/IndicTrans2#multilingual-translation-models) as a part of this work.

Thank you!

**Comment by look4pritam** [comment]
I have used IndicTrans models and created RESt based service. We are using it for more than 6 months.
It is accurate than Google translate. 
Thanks for releasing dataset and models.

But it doesn't work for scientific words. These words are available with their corresponding translation in the form of dictionary. Some of these are specific to the organization.

I am planning to use these as golden dataset and train the model.
I can manage upto 8 A100 with 80GB memory. So it can be done in 20 days time.

Thanks for helping me out.

**Comment by jaygala24** [comment]
Hey

The models released as a part of this work are trained on BPCC which mostly contains bitext pairs of general news articles. So, there might be definitely scope for improving the models at translating technical and scientific terms. However, I would recommend you first fine-tune the released models and see if that resolves the issue or not. My only concern with training from scratch might be as effective due to the limited amount of scientific translation data compared to the BPCC dataset and the model might not improve at translating technical words as expected.

Let me know if you have any further questions.

**Comment by look4pritam** [comment]
Thanks for your suggestion. I will first try fine-tuning as suggested by you. If that is not working then only I will try for training from scratch.
